src/main/java/uk/ac/cam/cl/dtg/segue/configuration/SegueGuiceConfigurationModule.java

```diff
@@ -1241,21 +1246,21 @@ public static Reflections getReflectionsClass(final String pkg) {
    *
    * @return the list of context listener classes (these should all be singletons).
    */
-  public static Collection<Class<? extends ServletContextListener>> getRegisteredContextListenerClasses() {
+  public static Collection<Class<? extends ServletContextListener>> getRegisteredContextListenerClasses()
+      throws IOException, URISyntaxException, ClassNotFoundException {
 
     if (null == contextListeners) {
       contextListeners = Lists.newArrayList();
 
-      Set<Class<? extends ServletContextListener>> subTypes = getReflectionsClass("uk.ac.cam.cl.dtg.segue")
```

Comment:
Some context is being lost here (and in the line below) with the new implementation. Before these changes, you knew that the `subTypes` variable would be a set of classes that extend `ServletContextListener`, but now that information is gone. I think you should be able to modify the `getSubTypes` contract by using generics, so that it returns a set of classes that extend from the `parentClass` parameter.

Reply:
I did have a look into that previously but it was showing an error about the class types having incompatible bounds or something. I probably didn't get the application of generics quite right, I'll have to take another look...